store/search/ts.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/typesense/typesense-go/typesense"
+	tsApi "github.com/typesense/typesense-go/typesense/api"
+)
+
+type storeImpl struct {
+	client *typesense.Client
+}
+
+type IndexDocumentsOptions struct {
+	Action    string
+	BatchSize int
+}
+
+func (s *storeImpl) convertToInternalError(err error) error {
+	if e, ok := err.(*typesense.HTTPError); ok {
+		switch e.Status {
+		case http.StatusConflict:
+			return ErrDuplicateEntity
+		case http.StatusNotFound:
+			return ErrNotFound
+		}
+		return NewSearchError(e.Status, ErrCodeUnhandled, e.Error())
+	}
+
+	return err
+}
+
+func (s *storeImpl) DeleteDocuments(_ context.Context, table string, key string) error {
+	_, err := s.client.Collection(table).Document(key).Delete()
+	return s.convertToInternalError(err)
+}
+
+func (s *storeImpl) IndexDocuments(_ context.Context, table string, documents io.Reader, options IndexDocumentsOptions) (err error) {
```

Comment:
I think the closure is redundant and we should handle the errors:
```
func (s *storeImpl) IndexDocuments(_ context.Context, table string, documents io.Reader, options IndexDocumentsOptions) error {
	reader, err := s.client.Collection(table).Documents().ImportJsonl(documents, &tsApi.ImportDocumentsParams{
		Action:    &options.Action,
		BatchSize: &options.BatchSize,
	})
    if err != nil {
        return err
    }
    defer func() { ulog.E(reader.Close()) }()

	type resp struct {
		Code     int
		Document string
		Error    string
		Success  bool
	}

	var r resp
	res, err := io.ReadAll(reader)
    if err != nil {
        return err
    }
	if err := jsoniter.Unmarshal(res, &r); err != nil {
        return err
    }
	if len(r.Error) > 0 {
		return fmt.Errorf(r.Error)
	}

	return nil
}
```

Reply:
yeah closure is not needed here.